lib/options.py

```diff
@@ -228,6 +228,22 @@ def __init__(self):
                     Default: false.
                     """))
 
+        parser.add_argument(
+                "--static-build",
+                dest="static_build",
+                action='store_true',
+                default=False,
+                help=format_help(
+                    """
+                    Mark tarantool build as static:
+
+                    - tarantoolctl utility will be used from the $TEST_RUN_DIR;
+                    - tarantool binary will be search in the '/static-build` directory
+                    or you can set the custom path via the `STATIC_BINARY` env variable.
+
+                    Default: false.
+                    """))
```

Comment:
test-run already has a lot of code to choose tarantool executable automatically or manually.

First, test-run supports a usual development flow, when tests are run from a build directory. If it is the same as a source directory (so called in-source build, `cmake . && make -j`), everything works without any manual path tuning. There is `--builddir` for an out-of-source build (`cmake /path/to/source/dir && make -j`), it is integrated into `make test` in tarantool. So, it works pretty good too.

Next, if we need to run the testing against an externally installed tarantool, test-run search it in `PATH` (if the locally built one is not found). Sometimes it is confusing in context of running of the tarantool's test suite, see #249. Sometimes, convenient. But anyway: there is a way to run the testing against arbitrary tarantool executable. And it uses pretty standard Unix approach with the `PATH` variable.

So, I'm in doubt that we need another way to pass the executable path.

And I don't undestand, to be honest, from where `/static-build` default comes. It looks as quite unusual root directory.

I suspect that test-run has all the needed executable search functionality and maybe even more than needed.

(And, again, it is not related anyhow to how tarantool links needed libraries.)

Reply:
Thanks for the comment! Indeed, my bad.

The problem was that I thought of the static assembly as a **"custom"** approach with **"unique parameters"**. It turned out that this was not the case.

We want to add an additional parameter `--executable` for the binary path (it seems to be convenient).